### PR TITLE
refactor/improve monitor mode selection

### PR DESCRIPTION
**Pull request by ikalco**
#### Describe your PR, what does it fix/add?
aims to 
* always ensure a mode is selected, if requested fails still set one, i might add an error notification
* clarify mode selection code
* consistent logs

also fixes format selection trying to use INVALID

#### Is it ready for merging, or does it need work?
yes probably
maybe some other people should test too, idk who tho

### CODE REVIEW

**Comment by ikalco** [comment]
this is ready for review
just need figure out how to test custom modes and do that

**Comment by ikalco** [comment]
rebased + fixed
tested highrr, highres, requested mode, and custom mode, works for me :tm:
also fallbacks are working

**Comment by vaxerski** [comment]
ping me after xmas

**Comment by ikalco** [comment]
@vaxry its after xmas :)

File: src/helpers/Monitor.cpp [hunk 1]
```diff
@@ -361,6 +363,380 @@ void CMonitor::onDisconnect(bool destroy) {
...
+
+    // try requested as custom mode jic it works
+    if (!success && RULE->resolution != Vector2D() && RULE->resolution != Vector2D(-1, -1) && RULE->resolution != Vector2D(-1, -2)) {
+        auto        refreshRate = output->getBackend()->type() == Aquamarine::eBackendType::AQ_BACKEND_DRM ? RULE->refreshRate * 1000 : 0;
+        auto        mode        = makeShared<Aquamarine::SOutputMode>(Aquamarine::SOutputMode{.pixelSize = RULE->resolution, .refreshRate = refreshRate});
+        std::string modeStr     = std::format("{:X0}@{:.2f}Hz", mode->pixelSize, mode->refreshRate / 1000.f);
+
+        output->state->setCustomMode(mode);
+
+        if (!state.test()) {
```
Comment (vaxerski) [inline src/helpers/Monitor.cpp:561]: lol man put a {} around a short if

**Review by vaxerski — COMMENTED** [review]
nice work in general. I like the approach.

File: src/helpers/Monitor.cpp (hunk 1 above)
Comment (ikalco) [reply src/helpers/Monitor.cpp]: :(

**Review by vaxerski — APPROVED** [review]

**Pull request merged**